registry/api/errcode/doc.go

```diff
@@ -0,0 +1,64 @@
+// Package errcode defines an error management package.
+//
+// The `errcode` package allows for each error to be typed and include all
+// information necessary to have further processing done on them if necessary.
+//
+// Use of this package is defined by the following flow:
+// * Each error is registered with the errcode package via the `Register()`
+// function. The `Register()` function takes a `group` name and an
+// `ErrorDescriptor` structure. The `group` name allows for errors to be
+// associated with a particular component, or any other grouping mechanism
+// that may have meaning to the code registering the error.  The
+// `ErrorDescriptor` describes the error itself. See below for more
+// information. The `Register()` function will return an `ErrorCode`
+// that uniquely identifies the register error.
+//
+// * Once an error is registered, the return `ErrorCode` can be used just like
+// any other golang `error` type.
+//
+// * If a particular error needs to have additional information or processing
+// performed then the `WithArgs()` and `WithDetails()` functions are available.
+// The `WithArgs()` function allows for the code generating the error to
+// specify the substitution values of the `%s` variables in the error's
+// message. `WithDetails()` allows for the specification of any additional
+// information that may need to be provided to the end-user for this particular
+// error.  In both cases, the functions will return an `Error` resource
+// which extends the `ErrorCode` resource with the additonal information.
+//
+// The package consists of three main resource types:
+//
+// * ErrorCode
+// ErrorCode is a unique (numerical) identifier for a particular error
```

Comment:
On second thought, this section is quite redundant: http://godoc.org/github.com/docker/distribution/registry/api/errcode#ErrorDescriptor.

This package overview should focus on use case and best practices. Examples:

1. Show that ErrorCodes should be used at the API level, at not the package level. This firewalls internal errors from errors providing to clients, reducing the potential of data leaks.
2. Show how to properly register errors.

Checkout [distribution/context](http://godoc.org/github.com/docker/distribution/context) for an example.